File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

public class Solution {
            public int RemoveNthFromEnd(ListNode curr, ListNode prev, int n)
            {
                if (curr == null) return 0;
                var nFromEnd = RemoveNthFromEnd(curr.next, curr, n) + 1;

                if (n == nFromEnd)
                {
                    if (prev is not null) prev.next = curr.next;
                    curr.next = null;
                }

                return nFromEnd;
            }

            public ListNode RemoveNthFromEnd(ListNode head, int n)
            {
                var dummy = new ListNode();
                dummy.next = head;

                RemoveNthFromEnd(dummy, null, n);
                return dummy.next;
            }
}

File: Data Structures & Algorithms/reverse-a-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head: return None
        if not head.next: return head

        stack = [head]
        curr = head.next
        head.next = None
        
        while curr.next:
            stack.append(curr)
            curr = curr.next
        
        newHead = curr

        while stack:
            curr.next = stack.pop()
            curr = curr.next
            curr.next = None
        

        return newHead




        
        